### ISSUE TITLE
Getting 'OutOfMemoryError' from 'test-lib' project when running androidTest task

### ISSUE THREAD

**Issue by DanDevine** [jaredsburrows/android-gradle-java-app-template#18]
I'm not 100% sure what's happening here.  I get the following message when building from Gradle 2.10 without the wrapper.  You can see the desired task was 'connectedDebugAndroidTest' and I wouldn't think that the 'test-lib' project should have anything that needs running in this case.

I found this in the unmodified project while trying to track down a similar error in a project that I modeled after the template.

UNEXPECTED TOP-LEVEL ERROR:
java.lang.OutOfMemoryError: GC overhead limit exceeded
	at java.util.regex.Pattern$BnM.optimize(Pattern.java:5408)
	at java.util.regex.Pattern.compile(Pattern.java:1709)
	at java.util.regex.Pattern.<init>(Pattern.java:1351)
	at java.util.regex.Pattern.compile(Pattern.java:1054)
	at java.lang.String.replace(String.java:2226)
	at com.android.dx.cf.direct.ClassPathOpener.compareClassNames(ClassPathOpener.java:197)
	at com.android.dx.cf.direct.ClassPathOpener.access$000(ClassPathOpener.java:37)
	at com.android.dx.cf.direct.ClassPathOpener$3.compare(ClassPathOpener.java:252)
	at com.android.dx.cf.direct.ClassPathOpener$3.compare(ClassPathOpener.java:250)
	at java.util.TimSort.countRunAndMakeAscending(TimSort.java:356)
	at java.util.TimSort.sort(TimSort.java:230)
	at java.util.Arrays.sort(Arrays.java:1435)
	at java.util.Collections.sort(Collections.java:230)
	at com.android.dx.cf.direct.ClassPathOpener.processArchive(ClassPathOpener.java:250)
	at com.android.dx.cf.direct.ClassPathOpener.processOne(ClassPathOpener.java:166)
	at com.android.dx.cf.direct.ClassPathOpener.process(ClassPathOpener.java:144)
	at com.android.dx.command.dexer.Main.processOne(Main.java:672)
	at com.android.dx.command.dexer.Main.processAllFiles(Main.java:574)
	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:311)
	at com.android.dx.command.dexer.Main.run(Main.java:277)
	at com.android.dx.command.dexer.Main.main(Main.java:245)
	at com.android.dx.command.Main.main(Main.java:106)

:test-lib:transformClassesWithDexForDebugAndroidTest FAILED

FAILURE: Build failed with an exception.

Within the top level 'build.gradle' file, you have the following line:
.....
    dependencies {
        // testing
        if (name.contains('test'))
.....

Is this supposed to match on ALL tests or just the unit test task?

Thanks, DD

**Comment by jaredsburrows** [comment]
@DanDevine Wow. Thanks. Good find. For some reason, `gradle tasks` does show `connectedDebugAndroidTest` for the `android-library`.

For now, you can do it like I do in the `travis.yml`: `gradle :app:connectedDebugAndroidTest`.

**Comment by DanDevine** [comment]
Really a newbie with Gradle, syntax, so I'm not sure what the 'name' field is checking on.  Is it the name of the Task?  Name of the Project?..

Would the check be better as (!name.equals('androidTest')) or perhaps (name.equals('unitTest'))?
Is there a way to debug the gradle script and see the variables/properties while running it?

Not trolling, I really don't know Gradle well enough.

**Comment by jaredsburrows** [comment]
`androidTest` is a task. I am simply filtering `dependencies` here. That is all.

**Comment by jaredsburrows** [comment]
@DanDevine Are you still having trouble with this?

**Comment by DanDevine** [comment]
No, seemed to be an issue with the size of the application and included
libraries.  I trimmed all the redundant bits, set my heap size larger for
dexing and didn't see the problem any longer.

It also seems that the tools version 1.50 supports JUnit / Robolectric
testing as build variant.  I switched to that recently, only issue there is
the dreaded 'stub' error, which requires moving the JUnit dependency before
the Android SDK library so that the proper JUnit version is included.

Thanks, can update with more details if interested.

DD
